test/functional/test_framework/wallet.py

```diff
@@ -119,12 +119,21 @@ def _bulk_tx(self, tx, target_weight):
     def get_balance(self):
         return sum(u['value'] for u in self._utxos)
 
-    def rescan_utxos(self):
+    def rescan_utxos(self, *, include_mempool=True):
         """Drop all utxos and rescan the utxo set"""
         self._utxos = []
         res = self._test_node.scantxoutset(action="start", scanobjects=[self.get_descriptor()])
         assert_equal(True, res['success'])
+        spents = set()
+        for txid in self._test_node.getrawmempool():
+            tx = self._test_node.getrawtransaction(txid=txid, verbose=True)
+            if include_mempool:
+                self.scan_tx(tx)
```

Comment:
You will need to sort the transactions topologically, see also `BlockAssembler::SortForBlock`